r-package/grf/R/lm_forest.R

```diff
@@ -0,0 +1,354 @@
+                                treatment = W.centered,
+                                sample.weights = sample.weights)
+  args <- list(num.trees = num.trees,
+               clusters = clusters,
+               samples.per.cluster = samples.per.cluster,
+               gradient.weights = gradient.weights,
+               sample.fraction = sample.fraction,
+               mtry = mtry,
+               min.node.size = min.node.size,
+               honesty = honesty,
+               honesty.fraction = honesty.fraction,
+               honesty.prune.leaves = honesty.prune.leaves,
+               alpha = alpha,
+               imbalance.penalty = imbalance.penalty,
+               stabilize.splits = stabilize.splits,
+               ci.group.size = ci.group.size,
+               compute.oob.predictions = compute.oob.predictions,
+               num.threads = num.threads,
+               seed = seed)
+
+  forest <- do.call.rcpp(multi_causal_train, c(data, args))
+  class(forest) <- c("lm_forest", "grf")
+  forest[["seed"]] <- seed
+  forest[["ci.group.size"]] <- ci.group.size
+  forest[["X.orig"]] <- X
+  forest[["Y.orig"]] <- Y
+  forest[["W.orig"]] <- W
+  forest[["Y.hat"]] <- Y.hat
+  forest[["W.hat"]] <- W.hat
+  forest[["clusters"]] <- clusters
+  forest[["equalize.cluster.weights"]] <- equalize.cluster.weights
+  forest[["sample.weights"]] <- sample.weights
+  forest[["has.missing.values"]] <- has.missing.values
+
+  forest
+}
+
+#' Predict with a lm forest
+#'
+#' Gets estimates of \eqn{h_k(x)}, k = 1..K in the conditionally linear model
+#' \eqn{Y = c(x) + h_1(x)W_1 + ... + h_K(x)W_K}, for a target sample X = x.
+#'
+#' @param object The trained forest.
+#' @param newdata Points at which predictions should be made. If NULL, makes out-of-bag
+#'                predictions on the training set instead (i.e., provides predictions at
+#'                Xi using only trees that did not use the i-th training example). Note
+#'                that this matrix should have the number of columns as the training
+#'                matrix, and that the columns must appear in the same order.
+#' @param num.threads Number of threads used in training. If set to NULL, the software
+#'                    automatically selects an appropriate amount.
+#' @param estimate.variance Whether variance estimates for hat{h_k}(x) are desired
+#'                          (for confidence intervals). This option is currently
+#'                          only supported for univariate outcomes Y.
+#' @param ... Additional arguments (currently ignored).
+#'
+#' @return A list with elements `predictions`: a 3d array of dimension [num.samples, K, M] with
+#' predictions for regressor W, for each outcome 1,..,M (singleton dimensions in this array can
+#' be dropped by passing the `drop` argument to `[`, or with the shorthand `$predictions[,,]`),
+#'  and optionally `variance.estimates`: a matrix with K columns with variance estimates.
+#'
+#' @examples
```

Comment:
Does it make sense to duplicate the example? Why not just say: See lm_forest for example usage

Reply:
Yes that's just the template for all forests + predict.forest